src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -388,6 +388,11 @@ namespace SurfaceGeometry {
                 if (state.dataSurface->ShadingSurfaceFirst == -1) state.dataSurface->ShadingSurfaceFirst = SurfNum;
                 state.dataSurface->ShadingSurfaceLast = SurfNum;
             }
+            if ((state.dataSurface->Surface(SurfNum).HeatTransSurf && state.dataSurface->Surface(SurfNum).ExtSolar) ||
```

Comment:
Seems ripe for `auto &thisSurface = state.dataSurface->Surface(SurfNum);` inside this loop.